pkg/metrics/metrics.go

```diff
@@ -73,6 +75,13 @@ const (
 func NewServerMetrics() *ServerMetrics {
 	return &ServerMetrics{
 		metrics: map[string]prometheus.Collector{
+			backupCountGauge: prometheus.NewGauge(
```

Comment:
I think this could be a `CounterVec` instead of a gauge. That way it can just be incremented with a call to `Inc()` inside the function.